SplitBrainTest.when_splitBrainProtectionDisabled_then_jobRunsTwiceAndAgainOnceAfterHeal failure

```
java.lang.AssertionError: expected:<10> but was:<15>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.jet.core.SplitBrainTest.lambda$when_splitBrainProtectionDisabled_then_jobRunsTwiceAndAgainOnceAfterHeal$37(SplitBrainTest.java:381)
	at com.hazelcast.jet.core.JetSplitBrainTestSupport.testSplitBrain(JetSplitBrainTestSupport.java:129)
	at com.hazelcast.jet.core.SplitBrainTest.when_splitBrainProtectionDisabled_then_jobRunsTwiceAndAgainOnceAfterHeal(SplitBrainTest.java:384)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

https://hazelcast-l337.ci.cloudbees.com/job/jet-pr-builder/com.hazelcast.jet$hazelcast-jet-core/3021/testReport/junit/com.hazelcast.jet.core/SplitBrainTest/when_splitBrainProtectionDisabled_then_jobRunsTwiceAndAgainOnceAfterHeal/

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JetSplitBrainTestSupport.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JetSplitBrainTestSupport.java
@@ -86,6 +86,7 @@ public abstract class JetSplitBrainTestSupport extends JetTestSupport {
         jetConfig.getInstanceConfig().setCooperativeThreadCount(PARALLELISM);
         jetConfig.getHazelcastConfig().setProperty(GroupProperty.MERGE_FIRST_RUN_DELAY_SECONDS.getName(), "5");
         jetConfig.getHazelcastConfig().setProperty(GroupProperty.MERGE_NEXT_RUN_DELAY_SECONDS.getName(), "5");
+        jetConfig.getInstanceConfig().setUpscaleDelayMillis(3000);
         onJetConfigCreated(jetConfig);
         return jetConfig;
     }

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/SplitBrainTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/SplitBrainTest.java
@@ -363,22 +363,23 @@ public class SplitBrainTest extends JetSplitBrainTestSupport {
             MockPS processorSupplier = new MockPS(StuckProcessor::new, clusterSize);
             DAG dag = new DAG().vertex(new Vertex("test", processorSupplier));
             jobRef[0] = instances[0].newJob(dag, new JobConfig().setSplitBrainProtection(false));
-            assertOpenEventually(StuckProcessor.executionStarted);
+            assertTrueEventually(() -> assertEquals("initCount", clusterSize, MockPS.initCount.get()), 10);
+            assertOpenEventually("executionStarted", StuckProcessor.executionStarted);
         };
 
         BiConsumer<JetInstance[], JetInstance[]> onSplit = (firstSubCluster, secondSubCluster) -> {
             Job jobRef1 = firstSubCluster[0].getJob(jobRef[0].getId());
             Job jobRef2 = secondSubCluster[0].getJob(jobRef[0].getId());
-            assertNotNull(jobRef1);
-            assertNotNull(jobRef2);
-            assertTrueEventually(() -> assertEquals(RUNNING, jobRef1.getStatus()), 10);
-            assertTrueEventually(() -> assertEquals(RUNNING, jobRef2.getStatus()), 10);
-            assertEquals(clusterSize * 2, MockPS.initCount.get());
+            assertNotNull("jobRef1", jobRef1);
+            assertNotNull("jobRef2", jobRef2);
+            assertTrueEventually(() -> assertEquals("job not running on subcluster 1", RUNNING, jobRef1.getStatus()), 10);
+            assertTrueEventually(() -> assertEquals("job not running on subcluster 2", RUNNING, jobRef2.getStatus()), 10);
+            assertEquals("initCount", clusterSize * 2, MockPS.initCount.get());
         };
 
         Consumer<JetInstance[]> afterMerge = instances -> {
-            assertTrueAllTheTime(() -> assertEquals(RUNNING, jobRef[0].getStatus()), 5);
-            assertEquals(clusterSize * 2, MockPS.initCount.get());
+            // this assert will hold after the job scales up
+            assertTrueEventually(() -> assertEquals(clusterSize * 3, MockPS.initCount.get()), 20);
         };
 
         testSplitBrain(firstSubClusterSize, secondSubClusterSize, beforeSplit, onSplit, afterMerge);

